modules/videostab/include/opencv2/videostab/frame_source.hpp

```diff
@@ -86,6 +86,23 @@ class CV_EXPORTS VideoFileSource : public IFrameSource
     Ptr<IFrameSource> impl;
 };
 
+class CV_EXPORTS MaskVideoFileSource : public IFrameSource
```

Comment:
No need to copy implementation of VideoFileSource:
- try to derive this class from `VideoFileSource`
- or better try to make proxy over any other IFrameSource:
  * ctor: `MaskFrameSource(const Ptr<IFrameSource>& source);`
  * usage: `MaskFrameSource(makePtr<VideoFileSource>(...));`

Also threshold value should vary in case of conditions (nightly frames are darker). Or user want to avoid lighter areas (sky region). There are too many cases really.
There is no "rocket science" to implement it, but too many application specific details which OpenCV is not able to cover (and should not - lets keep OpenCV as simple as possible).
So it is better to let user to implement masking by yourself and remove it from OpenCV modules code (you can still put this implementation in samples, see notes above).

Reply:
Updated the corresponding part. Used a proxy to VideoFileSource (i.e. the second recommended option). Used a callback too generalize the functionality of maskframesource